microsoft_atp/komand_microsoft_atp/triggers/get_alert_matching_key/trigger.py

```diff
@@ -14,21 +14,47 @@ def __init__(self):
                 output=GetAlertMatchingKeyOutput())
 
     def run(self, params={}):
-        """Run the trigger"""
-        while True:
-            key = params["key"]
-            value = params["value"]
-
-            matching_alerts = self.connection.get_alerts_by_key_value(key, value)
+        alert_key = params.get(Input.KEY)
+        alert_value = params.get(Input.VALUE)
 
-            if(len(matching_alerts) > 0):
-                self.send({"results": matching_alerts})
+        frequency = params.get("frequency", 10)
 
-            time.sleep(params.get("interval", 5))
+        # Set a baseline for the time to start looking for alerts.
+        initial_results = self.connection.get_all_alerts(query_parameters="?$orderby=alertCreationTime+desc&$top=1")
+        all_results = initial_results.json()
 
-    def test(self):
-        # This will raise an exception for failure cases
-        self.connection.test()
+        if len(all_results.get("value")):
```

Comment:
On second thought, you could simplify lines 26-31 with:
```
try:
    most_recent_result = all_results["value"][0]
    most_recent_result.get("alertCreationTime")
except (KeyError, IndexError):
    self.logger.info("No current alerts found, setting time to start looking to 2010-10-01.")
    most_recent_time_string = "2010-01-01T00:00:00.000000Z"  # We don't have any alerts yet
```

Reply:
I'd rather do an if there. I try to avoid using "try / except" logic for regular behavior.